src/ArraySubsetAsserts.php

```diff
@@ -7,12 +7,10 @@
 use ArrayAccess;
 use DMS\PHPUnitExtensions\ArraySubset\Constraint\ArraySubset;
 use Exception;
```

Comment:
These conditions were introduced in PR #65 in response to PR #50 and have a function (prevent the PHPUnit deprecation warning being thrown when using PHPUnit 8.x),

Using the generic PHP native `InvalidArgumentException` might break (external) tests using the polyfill, which rely on the PHPUnit native behaviour (which the current code closely emulates).

I understand changes may be needed to this logic toggle to allow for PHPUnit 10.0, but in that case, IMO, that should be handled in an additional toggle, not by breaking the existing functionality.

Reply:
Thinking this over some more.... I can imagine that a choice could be made to no longer use the PHPUnit native InvalidArgumentException/Helper.

In that case, I would recommend:
1. For this change to be annotated as a breaking change in the changelog, albeit one which will affect very few users.
2. To introduce an `InvalidArgumentException` class native to this package to handle the creation of the message.

Reply:
1. Mentioning this in the changelog sounds good - I would add it, but found that you do not maintain it as a document?
2. So add `class InvalidArgumentException extends \InvalidArgumentException`? I can make that change, but honestly see no benefit. Why not keep things as simple as possible?

Reply:
Is the issue here that PHPUnit 10 removes the `PHPUnit\Util\InvalidArgumentHelper` class?
I believe in that case that we should indeed throw a generic `\InvalidArgumentException` but only if the Helper is not present.